pkg/skaffold/gcbreposv2/repo_resolver.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2024 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcbreposv2
+
+import (
+	"context"
+	"fmt"
+
+	cloudbuild "cloud.google.com/go/cloudbuild/apiv2"
+	cloudbuildpb "cloud.google.com/go/cloudbuild/apiv2/cloudbuildpb"
+	"github.com/googleapis/gax-go/v2"
+)
+
+type cloudBuildRepoClient interface {
+	GetRepository(ctx context.Context, req *cloudbuildpb.GetRepositoryRequest, opts ...gax.CallOption) (*cloudbuildpb.Repository, error)
+	FetchReadToken(ctx context.Context, req *cloudbuildpb.FetchReadTokenRequest, opts ...gax.CallOption) (*cloudbuildpb.FetchReadTokenResponse, error)
+	Close() error
+}
+
+var RepositoryManagerClient = repositoryManagerClient
+
+func GetRepoInfo(ctx context.Context, gcpProject, gcpRegion, gcpConnectionName, gcpRepoName string) (repoURI string, readAccessToken string, err error) {
+	cbRepoRef := fmt.Sprintf("projects/%v/locations/%v/connections/%v/repositories/%v", gcpProject, gcpRegion, gcpConnectionName, gcpRepoName)
+	cbClient, err := RepositoryManagerClient(ctx)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create repository manager client: %w", err)
+	}
+	defer cbClient.Close()
+
+	repoURI, err = getrepoURI(ctx, cbClient, cbRepoRef)
```

Comment:
just want to call out, the returned uri has protocol, when we construct the url for `git` we need to remove `https://`

Reply:
Yep! Thanks for head-up. I moved the logic that builds the git clone URI to this file so we have all in the same place.